### PR TITLE
✍ Added kickassanime presence

### LINKED ISSUE #834: Kickassanime.rs | www.kickassanime.rs
> **Prerequisites and essential questions**
> 
> * [x]  Is it a popular site?
> * [x]  Is the website older than 2 months?
> * [ ]  Is the site locked to a specific country/region?
> * [ ]  Is the site a paid service? (e.g. Netflix, Hulu)
> * [ ]  Does the website feature NSFW content? (e.g. porn, etc...)
> * [ ]  Are you a donator/patron?
> * [x]  Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?
> 
> **What's your Discord username?** Ethical Hacker #4166
> 
> **What's the name of the service?** https://www.kickassanime.rs/ - kickassanime
> 
> **What should the Presence display?** The name of anime, episode and what your browsing through. Similar to YouTube or other anime sites.
> 
> **If possible, please provide a logo for the service (512x512 minimum)** [their official logo](https://www.kickassanime.rs/images/logo.png)

> **Extra Info** This site is Very Very fast as compared to kissanime, gogoanime and 9anime most of the time this is the first site i see ep release and then other sites (of course except paid sites like netflix,etc)

take time as long as you want but be sure to add it Thanks for reading though...

### PR DESCRIPTION
**Pull request by StrikerFRFX**
All screenshots are of mapped out pages
I chose not to add the settings page in but I can add it on request
Timestamps work as expected

guxKvGVs2H
bsDZgpex99
E5IAXb73UF
RVmD6876Zk
aUYlB18fEr
DskKDzlQNl
CyZrIMxBQk
KiDzzFQpx0
9Dze8uxAxH
FU5SaUoGU6

Resolves #834

### CODE REVIEW

File: websites/K/Kickassanime/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "author": {
+      "name": "Striker",
+      "id": "215509157837537280"
+    },
+    "service": "Kickassanime",
+    "description": {
+      "en": "Kickassanime is a streaming website where you can watch anime in dubbed formats and subbed formats."
+    },
+    "url": "kickassanime.rs",
+    "version": "1.3.6",
```
Comment (Bas950) [inline websites/K/Kickassanime/dist/metadata.json:11]: version 1.0.0 on release

File: websites/K/Kickassanime/presence.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+   * @param {Number} videoTime Current video time seconds
+   * @param {Number} videoDuration Video duration seconds
+   */
+  function getTimestamps(
+    videoTime: number,
+    videoDuration: number
+  ): Array<number> {
+    const startTime = Date.now(), endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+    return [Math.floor(startTime / 1000), endTime];
+  }
```
Comment (Bas950) [inline websites/K/Kickassanime/presence.ts:38]: use the presence.getTimestamps instead

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/K/Kickassanime/dist/metadata.json (hunk 1 above)
Comment (StrikerFRFX) [reply websites/K/Kickassanime/dist/metadata.json:11]: Oops typo from copying existing metadata files sry

File: websites/K/Kickassanime/presence.ts (hunk 2 above)
Comment (StrikerFRFX) [reply websites/K/Kickassanime/presence.ts:38]: Okay

File: websites/K/Kickassanime/presence.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    currentAnimeWatching: Array<string>,
+    currentAnimeTitle: string,
+    currentAnimeEpisode: string,
+    isMovie: boolean = null;
+  
+  /**
+   * Get Timestamps
+   * @param {Number} videoTime Current video time seconds
+   * @param {Number} videoDuration Video duration seconds
+   */
```
Comment (ririxidev) [inline websites/K/Kickassanime/presence.ts:31]: eee delete it?

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/K/Kickassanime/presence.ts (hunk 3 above)
Comment (StrikerFRFX) [reply websites/K/Kickassanime/presence.ts:31]: oopsies XD

**Review by Bas950 — APPROVED** [review]

**Review by ririxidev — APPROVED** [review]

**Pull request merged**